src/include/duckdb/common/serializer/binary_deserializer.hpp

```diff
@@ -45,13 +45,17 @@ class BinaryDeserializer : public FormatDeserializer {
 	struct State {
 		uint32_t expected_field_count;
 		idx_t expected_size;
+		field_id_t expected_field_id;
```

Comment:
Does `expected_field_count` mean that there are multiple fields? Or is this the size of one field (`expected_field_size` might be easier to understand)? And if multiple fields exist, why is there only one `expected_field_id`?

I am leaving these questions here to document my code-reading process a bit...